### ISSUE TITLE
An example where distribution is extended

### ISSUE THREAD

**Issue by cranmer** [diana-hep/carl-notebooks#5]
I assume that we can add a distribution without modifying the `carl` package, but I haven't checked. 
If so, an example notebook with that would be nice.

**Comment by glouppe** [comment]
Yes this is possible and quite straightforward (i really need to write documentation... !).

For the likelihood-free part (approximating ratios), the only requirement is that the object that represents the generator exposes a `rvs(n_samples)` method that returns an array of samples. 

For implementing an actual distribution, the requirement is to implement the `carl.distributions.DistributionMixin` interface. And for nice interactions with the theano world, one should inherit from `carl.distributions.TheanoDistribution`.
